src/main/java/net/sf/jabref/gui/preftabs/PreviewPrefsTab.java

```diff
@@ -1,160 +1,119 @@
+    private final DefaultListModel<Object> chosenModel = new DefaultListModel<>();
+
+    private final JList<Object> available = new JList<>(availableModel);
+    private final JList<Object> chosen = new JList<>(chosenModel);
+
+    private final JButton btnRight = new JButton("»");
+    private final JButton btnLeft = new JButton("«");
+    private final JButton btnUp = new JButton(Localization.lang("Up"));
+    private final JButton btnDown = new JButton(Localization.lang("Down"));
+
+
+    private final JTextArea layout = new JTextArea("", 1, 1);
+    private final JButton btnTest = new JButton(Localization.lang("Test"));
+    private final JButton btnDefault = new JButton(Localization.lang("Default"));
+    private final JScrollPane scrollPane = new JScrollPane(layout);
+
+
+    public PreviewPrefsTab() {
+        setupLogic();
+        setupGui();
+    }
+
+    private void setupLogic(){
+        chosen.getSelectionModel().addListSelectionListener(e -> {
+            boolean selectionEmpty = ((ListSelectionModel) e.getSource()).isSelectionEmpty();
+            btnLeft.setEnabled(!selectionEmpty);
+            btnDown.setEnabled(!selectionEmpty);
+            btnUp.setEnabled(!selectionEmpty);
         });
 
-        defaultButton2.addActionListener(e -> {
-            String tmp = layout2.getText().replace("\n", "__NEWLINE__");
-            PreviewPrefsTab.this.prefs.remove(JabRefPreferences.PREVIEW_1);
-            layout2.setText(PreviewPrefsTab.this.prefs.get(JabRefPreferences.PREVIEW_1).replace("__NEWLINE__", "\n"));
-            PreviewPrefsTab.this.prefs.put(JabRefPreferences.PREVIEW_1, tmp);
+        available.getSelectionModel()
+                .addListSelectionListener(e -> btnRight.setEnabled(!((ListSelectionModel) e.getSource()).isSelectionEmpty()));
+
+        btnRight.addActionListener(e -> {
+            for (Object s : available.getSelectedValuesList()) {
+                availableModel.removeElement(s);
+                chosenModel.addElement(s);
+            }
         });
 
-        testButton.addActionListener(e -> {
-            try {
-                PreviewPanel testPanel = new PreviewPanel(null, TestEntry.getTestEntry(), null, layout1.getText());
-                testPanel.setPreferredSize(new Dimension(800, 350));
-                JOptionPane.showMessageDialog(null, testPanel, Localization.lang("Preview"), JOptionPane.PLAIN_MESSAGE);
-            } catch (StringIndexOutOfBoundsException ex) {
-                LOGGER.warn("Parsing error.", ex);
-                JOptionPane.showMessageDialog(null,
-                        Localization.lang("Parsing error") + ": " + Localization.lang("illegal backslash expression")
-                                + ".\n" + ex.getMessage(),
-                        Localization.lang("Parsing error"), JOptionPane.ERROR_MESSAGE);
+        btnLeft.addActionListener(e -> {
+            for (Object s : chosen.getSelectedValuesList()) {
```

Comment:
Please rename, this applies to all the variables that are only one or two characters long.